builder/amazon/common/ssh.go

```diff
@@ -16,27 +17,29 @@ func SSHAddress(e *ec2.EC2, port int, private bool) func(multistep.StateBag) (st
 		for j := 0; j < 2; j++ {
 			var host string
 			i := state.Get("instance").(*ec2.Instance)
-			if i.VpcId != "" {
-				if i.PublicIpAddress != "" && !private {
-					host = i.PublicIpAddress
+			if *i.VPCID != "" {
+				if *i.PublicIPAddress != "" && !private {
+					host = *i.PublicIPAddress
 				} else {
-					host = i.PrivateIpAddress
+					host = *i.PrivateIPAddress
 				}
-			} else if i.DNSName != "" {
-				host = i.DNSName
+			} else if *i.PublicDNSName != "" {
```

Comment:
I think that if `i.VPCID`, `i.PublicIPAddress` or `i.PriviateIPAddress` are nil, you will get a panic here. Need to check `i.VpcId != nil` et. al. as well